components/experimental/src/duration/format.rs

```diff
@@ -710,19 +710,34 @@ mod tests {
             [
                 (0, 6, parts::YEAR),
                 (0, 6, icu_list::parts::ELEMENT),
+                (0, 2, icu_pattern::parts::PLACEHOLDER),
```

Comment:
Discussion: It shouldn't be public that DurationFormatter happens to use `icu_pattern` here. I think we should probably suppress these parts. Two ways to do that:

1. Override the sink to throw out parts with category `pattern`
2. Add API to `icu_pattern` to not write out the parts

Option 2 seems cleaner, but it may complicate the traits, which are already complicated.

@Manishearth

Reply:
I definitely think that real formatting code should basically never be emitting PLACEHOLDER here. icu_pattern is an internal abstraction, unlike the concept of a list. I would rather fix this at the source than at the sink.


Which traits? I think for Option 2 we wouldn't go through the traits: this wouldn't be part of the Writeable impl for icu_pattern, but rather a helper function that allows you to write a pattern without introducing further parts.



Thoughts?

Reply:
Discussion:

- Add `interpolate_with_pattern_parts` since there are 2 behaviors which could both make sense.
- Or, just make the inner writeable write the placeholder part if it wants to, and remove the placeholder part from the trait. Open question on how to filter out the literal part.